application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelStatus.java

```diff
@@ -0,0 +1,205 @@
+     */
+    public @NotNull String getName() {
+        return name;
+    }
+
+    private void setName(@NotNull final String name) {
+        this.name = Objects.requireNonNull(name);
+    }
+
+    /**
+     * Method used to keep the channel name up to date with the {@link JDA}. This method is not
+     * called automatically. Manually update before using the value.
+     * <p>
+     * The recommended value to use is {@link TextChannel#getAsMention()}
+     * <p>
+     * This method is called in multithreaded context, however the value is not expected to change
+     * regularly and will not break anything if it is incorrect for a read or two, so it has not
+     * been made thread safe.
+     *
+     * @param guild the {@link Guild} that the channel belongs to, to retrieve its name from.
+     */
+    public void updateChannelName(@NotNull final Guild guild) {
+        GuildChannel channel = guild.getGuildChannelById(channelId);
+        if (channel == null) {
+            throw new IllegalArgumentException(
+                    "The guild passed in '%s' is not related to the channel this status is for: %s"
+                        .formatted(guild.getName(), this));
+        }
+        if (channel instanceof TextChannel textChannel) {
+            setName(textChannel.getAsMention());
+        } else
+            throw new IllegalStateException(
+                    "This channel status was created with the id for a non-text-channel and status cannot be monitored: '%s'"
+                        .formatted(channelId));
+    }
+
+    /**
+     * Method to set the channel status to busy, a user id is passed in to keep track of the current
+     * user requesting help. This id will be used to confirm that the author is satisfied with the
+     * channel being marked as free.
+     * <p>
+     * This functionality is not yet implemented so the id can be anything atm. Also note that on
+     * reboot the bot does not currently search for the author so the first time its marked as free
+     * there will be no confirmation.
+     * 
+     * @param userId the id of the user who changed the status to 'busy'
+     */
+    public void setBusy(final long userId) {
+        if (isBusy == FREE) {
+            this.isBusy = BUSY;
+            this.userId = userId;
+        }
+    }
+
+    /**
+     * Method to set the channel status to free, the user id of the previous help requester is not
+     * overwritten by this method. So until another user changes the status to busy the old value
+     * will remain.
+     * <p>
+     * The value will be 0 until the first time that the status is changed from free to busy.
+     * <p>
+     * This functionality is not yet implemented so the id can be anything atm.
+     */
+    public void setFree() {
+        if (isBusy == BUSY) {
+            isBusy = FREE;
+        }
+    }
+
+    // todo should I overload equals with equals(long) so that a Set may be used instead of a Map
```

Comment:
`// TODO` not `// todo`
might also want to move the todo inside the method instead of above it, so that when the method gets moved around, the todo is still inside it.

Reply:
fixed

Reply:
could you move it **inside the method body**? otherwise method re-ordering might move the method elsewhere but leave the comment there in the void.

Reply:
fixed

Reply:
its still outside, did you forget to push?

![](https://i.imgur.com/c8Yt8M4.png)

Reply:
its pushed